src/Simplex_tree/concept/FiltrationValue.h

```diff
@@ -5,19 +5,106 @@
  *    Copyright (C) 2014 Inria
  *
  *    Modification(s):
+ *      - 2024/08 Hannah Schreiber: Update of the concept after several additions to the Simplex tree.
  *      - YYYY/MM Author: Description of the modification
  */
 
 /** \brief Value type for a filtration function on a cell complex.
-  *
-  * A <EM>filtration</EM> of a cell complex (see FilteredComplex) is
-  * a function \f$f:\mathbf{K} \rightarrow \mathbb{R}\f$ satisfying \f$f(\tau)\leq 
-  * f(\sigma)\f$ whenever \f$\tau \subseteq \sigma\f$. Ordering the simplices 
-  * by increasing filtration values (breaking ties so as a simplex appears after 
-  * its subsimplices of same filtration value) provides an indexing scheme 
-  * (see IndexingTag).
-  */
-  struct FiltrationValue {
-    /** \brief Operator < is a StrictWeakOrdering. */
-    bool operator<(FiltrationValue f1, FiltrationValue f2);
-  };
+ *
+ * Needs to implement `std::numeric_limits<FiltrationValue>::has_infinity`,
+ * `std::numeric_limits<FiltrationValue>::infinity()` and `std::numeric_limits<FiltrationValue>::max()`.
+ * But when `std::numeric_limits<FiltrationValue>::has_infinity` returns `true`,
+ * `std::numeric_limits<FiltrationValue>::max()` can simply throw when called, as well as,
+ * `std::numeric_limits<FiltrationValue>::infinity()` if `std::numeric_limits<FiltrationValue>::has_infinity`
+ * returns `false`.
+ *
+ * A <EM>filtration</EM> of a cell complex (see FilteredComplex) is
+ * a function \f$f:\mathbf{K} \rightarrow \mathbb{R}\f$ satisfying \f$f(\tau)\leq
+ * f(\sigma)\f$ whenever \f$\tau \subseteq \sigma\f$. Ordering the simplices
+ * by increasing filtration values (breaking ties so as a simplex appears after
+ * its subsimplices of same filtration value) provides an indexing scheme
+ * (see IndexingTag).
+ */
+struct FiltrationValue {
+  /**
+   * @brief Has to construct the default value of FiltrationValue.
+   */
+  FiltrationValue();
+  /**
+   * @brief Only necessary if `FiltrationValue(0)` does not correspond to an "empty" filtration value, or,
+   * if there is an ambiguity at compile time between several constructors if
+   * @ref Gudhi::simplex_tree::empty_filtration_value_t is passed.
+   * Should construct a representative of what is an "empty" object. E.g., 0 for a numerical value or {} for a vector.
+   * Two empty objects should evaluate to true if compared for equality.
+   */
+  FiltrationValue(Gudhi::simplex_tree::empty_filtration_value_t e);
```

Comment:
Should we add `explicit` (non-explicit also works, but it is only needed in explicit contexts)?

Reply:
Ok for `explicit`, it makes sense. In the end, the class author can do whatever they prefer anyway.